test/test-mc-rangeopts.c

```diff
@@ -104,15 +108,185 @@ static void test_mc_RangeOpts_to_FLE2RangeInsertSpec(_mongocrypt_tester_t *teste
+         .expect = RAW_STRING({"v" : {"v" : 0.0, "min" : 0.0, "max" : 100.0, "precision" : 1, "trimFactor" : 9}})},
+        {.desc = "tf bound check fails correctly for double with min, max, precision set (tf = 10, domain size < 2^10)",
+         .in = RAW_STRING(
+             {"trimFactor" : 10, "min" : 0.0, "max" : 100.0, "precision" : 1, "sparsity" : {"$numberLong" : "1"}}),
+         .v = RAW_STRING({"v" : 0.0}),
+         .expectError = "Trim factor (10) must be less than the total number of bits (10) used to represent any "
+                        "element in the domain."},
+
+        {.desc = "tf = 63 works for unbounded double (domain size = 2^64)",
+         .in = RAW_STRING({"trimFactor" : 63, "sparsity" : {"$numberLong" : "1"}}),
+         .v = RAW_STRING({"v" : 0.0}),
+         // note - when min and max are unset, they are added into the insert spec.
+         .expect = RAW_STRING({
+             "v" : {
+                 "v" : 0.0,
+                 "min" : {"$numberDouble" : "-1.7976931348623157081e+308"},
+                 "max" : {"$numberDouble" : "1.7976931348623157081e+308"},
+                 "trimFactor" : 63
+             }
+         })},
+        {.desc = "tf = 64 fails for unbounded double (domain size = 2^64))",
+         .in = RAW_STRING({"trimFactor" : 64, "sparsity" : {"$numberLong" : "1"}}),
+         .v = RAW_STRING({"v" : 0.0}),
+         .expectError = "Trim factor (64) must be less than the total number of bits (64) used to represent any "
+                        "element in the domain."},
+
+        {.desc = "tf bound check passes correctly for decimal with min, max, precision set (tf = 9, 2^9 < domain size "
+                 "< 2^10)",
+         .in = RAW_STRING({
+             "trimFactor" : 9,
+             "min" : {"$numberDecimal" : "0"},
+             "max" : {"$numberDecimal" : "100"},
+             "precision" : 1,
+             "sparsity" : {"$numberLong" : "1"}
+         }),
+         .v = RAW_STRING({"v" : {"$numberDecimal" : "0"}}),
+         .expect = RAW_STRING({
+             "v" : {
+                 "v" : {"$numberDecimal" : "0"},
+                 "min" : {"$numberDecimal" : "0"},
+                 "max" : {"$numberDecimal" : "100"},
+                 "precision" : 1,
+                 "trimFactor" : 9
+             }
+         })},
+        {.desc =
+             "tf bound check fails correctly for decimal with min, max, precision set (tf = 10, domain size < 2^10)",
+         .in = RAW_STRING({
+             "trimFactor" : 10,
+             "min" : {"$numberDecimal" : "0"},
+             "max" : {"$numberDecimal" : "100"},
+             "precision" : 1,
+             "sparsity" : {"$numberLong" : "1"}
+         }),
+         .v = RAW_STRING({"v" : {"$numberDecimal" : "0"}}),
+         .expectError = "Trim factor (10) must be less than the total number of bits (10) used to represent any "
+                        "element in the domain."},
+
+#if MONGOCRYPT_HAVE_DECIMAL128_SUPPORT
```

Comment:
Move `#if MONGOCRYPT_HAVE_DECIMAL128_SUPPORT` above the "tf bound check passes correctly for decimal" test so decimal tests do not run when libmongocrypt is not built with decimal128 support.

Reply:
Good catch, changed